src/js/components/MaskedInput/__tests__/MaskedInput-test.js

```diff
@@ -191,6 +191,55 @@ describe('MaskedInput', () => {
     expect(onChange).toHaveReturnedWith('aa!');
   });
 
+  test('should not enable to type beyond options via keyboard', async () => {
+    const onChange = jest.fn((event) => event.target.value);
+    const { getByTestId, container } = render(
+      <MaskedInput
+        data-testid="test-input"
+        id="item"
+        name="item"
+        mask={[
+          {
+            options: ['aaa', 'aba', 'abb'],
+            regexp: /\w$/,
+          },
+        ]}
+        onChange={onChange}
+      />,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+
+    fireEvent.change(getByTestId('test-input'), { target: { value: 'abbb' } });
+
+    expect(onChange).toHaveBeenCalled();
+    expect(onChange).toHaveReturnedWith('abb');
+  });
+
+  test('should enable to type beyond options when ignoreOptions', async () => {
+    const onChange = jest.fn((event) => event.target.value);
+    const { getByTestId, container } = render(
+      <MaskedInput
+        data-testid="test-input"
+        id="item"
+        name="item"
+        mask={[
+          {
+            ignoreOptions: true,
+            options: ['aaa', 'aba', 'abb'],
+            regexp: /\w$/,
+          },
+        ]}
+        onChange={onChange}
+      />,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+
+    fireEvent.change(getByTestId('test-input'), { target: { value: 'abbb' } });
```

Comment:
Same comment as above,
Can we use `userEvent` instead of `fireEvent` and `getByRole` instead of `getByTestId`?